### PR TITLE
Remove dd4hep warnings

### PR DESCRIPTION
**Pull request by asalzburger**
This is a cherry-pick plus add-ons and supersedes #118.

### CODE REVIEW

**Comment by asalzburger** [comment]
Hey @msmk0 and @paulgessinger - I get a `xerces` not found problem with `macOS` now, I haven't touched the general `CMakeLists.txt` file in this PR. Any idea ?

**Comment by paulgessinger** [comment]
> Hey @msmk0 and @paulgessinger - I get a `xerces` not found problem with `macOS` now, I haven't touched the general `CMakeLists.txt` file in this PR. Any idea ?

Looking into it

**Comment by asalzburger** [comment]
Ok, this has build failures in `macOS` (xerces-c Issue) and failing of the coverage, which should be fixed by #136

**Comment by asalzburger** [comment]
I will rebase this one and see if @paulgessinger 's coverage fix removes one failure.

**Comment by asalzburger** [comment]
Only macOS build failing, but that should be fixed with #141

**Comment by asalzburger** [comment]
Hi @msmk0 - this is now ready to go in.

It removes all dd4hep related warnings and puts some of the helper code in dedicated compilation units.

File: Examples/Detectors/DD4hepDetector/src/IBL/IBLSimpleBarrel_geo.cpp [hunk 1]
```diff
@@ -148,11 +148,11 @@ static Ref_t create_element(Detector& lcdd, xml_h xml, SensitiveDetector sens) {
           if (x_module.isSensitive()) {
             mod_vol.setSensitiveDetector(sens);
           }
-          int comp_num = 0;
+          int sens_comp_num = 0;
```
Comment (msmk0) [inline Examples/Detectors/DD4hepDetector/src/IBL/IBLSimpleBarrel_geo.cpp:151]: Should be `sensCompNum` to be consistent with the naming guidelines.

**Pull request merged**